Tensile/Source/client/main.cpp

```diff
@@ -602,21 +602,36 @@ int main(int argc, const char* argv[])
                             TimingEvents warmupStartEvents(warmupInvocations, eventCount);
                             TimingEvents warmupStopEvents(warmupInvocations, eventCount);
 
-                            for(int i = 0; i < warmupInvocations; i++)
+                            if (warmupInvocations > 0)
                             {
+                                // Do validation after first warmup
                                 listeners.preWarmup();
                                 if(gpuTimer)
                                     HIP_CHECK_EXC(adapter.launchKernels(kernels,
                                                                         stream,
-                                                                        warmupStartEvents[i],
-                                                                        warmupStopEvents[i]));
+                                                                        warmupStartEvents[0],
+                                                                        warmupStopEvents[0]));
                                 else
                                     HIP_CHECK_EXC(
                                         adapter.launchKernels(kernels, stream, nullptr, nullptr));
                                 listeners.postWarmup();
-                            }
+                                listeners.validateWarmups(inputs, warmupStartEvents, warmupStopEvents);
```

Comment:
You do not need to duplicate the code.
You can just put if (i==0) for validateWarmups.